plugins/fakePlugin.py

```diff
@@ -1,12 +1,17 @@
+from sys import path
+from os.path import dirname as dir
+path.append(dir(path[0]))
+from PluginSystem import PluginBase
+
 """
 Plugin used for testing.
 
 The class name must be the same as the file name.
 """
-class FakePlugin():
+class FakePlugin(PluginBase.PluginBase):
 
-    def __init__(self):
-        return
+    def init(self):
+        self.pluginEngine.RegisterCallback("realtime", "HelloWorld", self.HelloWorld, ["cameraStills", "otherStream"])
```

Comment:
we probably don't need this to pass the strings "realtime" and "HelloWorld." we're probably not gonna get to the historical engine (and that would be determined by plugin location anyway), and the HelloWorld string is redundant now that the method is being passed properly. Just function pointer and stream names is enough

Reply:
The dictionary needs a string to map to the array that holds the function pointer and the streams. So for this plugin, "HelloWorld" is the key to map to the values [self.HelloWorld, ["cameraStills", "otherStream"]]

Reply:
function pointers are hashable in python, why not just have it be the key?

Reply:
Didn't know function pointers are hashable! I'll update it to be the key!